lib/comm_common.cpp

```diff
@@ -432,12 +432,18 @@ int comm_coord(int dim)
 }
 
 
-inline bool isHost(const cudaPointerAttributes &attributes) {
-#if CUDA_VERSION >= 10000
-  return (attributes.type == cudaMemoryTypeHost);
-#else
-  return (attributes.memoryType == cudaMemoryTypeHost);
-#endif
+inline bool isHost(const void *buffer)
+{
+  CUmemorytype memType;
+  void *attrdata[] = {(void *)&memType};
+  CUpointer_attribute attributes[2] = {CU_POINTER_ATTRIBUTE_MEMORY_TYPE};
+  CUresult err = cuPointerGetAttributes(1, attributes, attrdata, (CUdeviceptr)buffer);
+  if (err != CUDA_SUCCESS) {
+    const char *str;
+    cuGetErrorName(err, &str);
+    errorQuda("cuPointerGetAttributes returned error %s", str);
+  }
+  return (memType != CU_MEMORYTYPE_DEVICE);
```

Comment:
While we currently only use device and host, for unified memory this would return true.

Reply:
Yes, for now I'll add an explicit check for unified memory and through an error if this is encountered.  While we may want to support unified memory on fields, we should *never* user unified memory for MPI buffers.